src/com/twitter/intellij/pants/service/project/resolver/PantsSourceRootsExtension.java

```diff
@@ -67,6 +77,72 @@ public void resolve(
 
       createContentRoots(moduleDataNode, targetInfo);
     }
+    installSyntheticModules(executor, projectDataNode, modules);
+  }
+
+  /**
+   * Takes a set of paths. Returns a new set without paths that are children of any other path in the set
+   * Ex. for [A, A/B, A/B/C, B, C/D] the result set is [A, B, C/D]
+   */
+  public static Set<String> removeChildren(Set<String> paths) {
+    if(paths.size() > 0 ) {
+      List<String> sorted = paths.stream().sorted().collect(Collectors.toList());
+      String current = sorted.get(0);
+      List<String> buf = new ArrayList<>();
+      buf.add(current);
+      for (String s : sorted.stream().skip(1).collect(Collectors.toList())) {
+        if (!s.startsWith(current)) {
+          current = s;
+          buf.add(s);
+        }
+      }
+      return new HashSet<>(buf);
+    }
+    return Collections.emptySet();
+  }
+
+  private void installSyntheticModules(
+    @NotNull PantsCompileOptionsExecutor executor,
+    @NotNull DataNode<ProjectData> projectDataNode,
+    @NotNull Map<String, DataNode<ModuleData>> modules
+  ) {
+    Path projectPath = Paths.get(executor.getBuildRoot().getAbsolutePath());
+    Set<String> targetSpecPaths =
+      executor.getOptions().getSelectedTargetSpecs().stream()
+        .map(x -> x.replaceFirst("/?:.*", ""))
```

Comment:
I copied the regex from here, it was assumed that the spec may be of the form `spec_path/:spec_target` I'm not sure if it's possible at all, but kept it just in case

https://github.com/scalameta/fastpass/blob/0e8092e8e650608aa879427e834cb1b1ab469501/fastpass/src/main/scala/scala/meta/internal/fastpass/pantsbuild/PantsConfiguration.scala#L93-L110

Reply:
Sounds good. Thanks!